src/TLuaInterpreter.cpp

```diff
@@ -10641,6 +10646,240 @@ int TLuaInterpreter::restartIrc(lua_State* L)
     return 1;
 }
 
+
+/** ttsSpeak( text )
+ *  Synthesizes text for reading.
+ */
+int TLuaInterpreter::ttsSpeak(lua_State* L)
+{
+    TLuaInterpreter::ttsBuild(0);
+    
+    string luaSpeechText = "";
+    if (!lua_isstring(L, 1)) {
+        lua_pushstring(L, "ttsSpeak: wrong argument type");
+        lua_error(L);
+        return 1;
+    } else {
+        luaSpeechText = lua_tostring(L, 1);
+    }
+    
+    QString text(luaSpeechText.c_str());
+    
+    speechUnit->say(text);
+    
+    return 1;
```

Comment:
Well if there isn't enough values on the "stack" Lua will just make _nil_ values as needed to match up with the `return` value - and as asking for more arguments from a lua function than it has to give you will also manufacture _nil_s as needed it is not particularly informative. For ___run-time___ errors the standard Lua way of doing things is to pop a _nil_ on the stack and follow that with a string message saying what is wrong, along the lines of:

    lua_pushnil(L);
    lua_pushstring(L, "You can't do that with that - or at least not right now with what you currently have!");
    return 2; // There are two things pushed on to the top of the lua stack that we want to return to the caller